### PR TITLE
Run script for nekbone barrier experiment

### LINKED ISSUE #1254: Create run script required for #1242
> estimate 3

Parent story: #1242 

Done criteria:
- Commit script to dev that support launching the runs for the experiment described in #1242
- Make sure that the profile names inserted in the report can be used by the visualization scripts in #1240 
- Make sure scripting supports different app configs and also different agents.
- Barrier regions are executed with a range of different CPU frequencies, and the frequency is captured in the profile name.

### PR DESCRIPTION
**Pull request by dianarg**
- Resolves #1254

### CODE REVIEW

File: integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    mach = machine.init_output_dir(output_dir)
+    min_freq = args.min_frequency
+    max_freq = args.max_frequency
+    step_freq = None
+    freqs = frequency_sweep.setup_frequency_bounds(mach, min_freq, max_freq, step_freq,
+                                                   add_turbo_step=True)
+    default_freq = max(freqs)
+    iterations = args.iterations
+
+    report_sig = ["CYCLES_THREAD@package", "CYCLES_REFERENCE@package",
```
Comment (cmcantalupo) [inline integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py]: Would be nice to have a report_singals() function in a base script for other run scripts to use

File: integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    # baseline run
+    targets = [launch_util.LaunchConfig(app_conf=baseline_app, agent_conf=None, name='base')]
+
+    # freq map runs
+    for freq in freqs:
+        rid = 'fma_{:.1e}'.format(freq)
+        options = {'FREQ_DEFAULT': default_freq,  # or use max or sticker from mach
+                   'HASH_0': barrier_hash,
+                   'FREQ_0': freq}
+        agent_conf = geopmpy.io.AgentConf('{}.config'.format(rid),
```
Comment (cmcantalupo) [inline integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py]: Be nice to have this loop that generates agent_conf list in a common script

File: integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                  "TIME@package", "ENERGY_PACKAGE@package"]
+    extra_cli_args += launch_util.geopm_signal_args(report_sig, [])
+
+    # baseline run
+    targets = [launch_util.LaunchConfig(app_conf=baseline_app, agent_conf=None, name='base')]
+
+    # freq map runs
+    for freq in freqs:
+        rid = 'fma_{:.1e}'.format(freq)
+        options = {'FREQ_DEFAULT': default_freq,  # or use max or sticker from mach
```
Comment (bgeltz) [inline integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py]: FREQ_UNCORE is the second policy param for the FMap agent according to ```$ geopmagent -a frequency_map```.  Are we intentionally letting it float here?

Comment (dianarg) [reply integration/experiment/energy_efficiency/run_barrier_frequency_sweep_nekbone.py]: Yes, I didn't see any obvious problems with it, but also planning to do some more analysis of the traces I have

**Review by cmcantalupo — APPROVED** [review]

**Pull request merged**